src/models/utci.js

```diff
@@ -0,0 +1,619 @@
+ * @param {number[]} rh - relative humidity, [%]
+ * @param {"SI"|"IP"} units - select the SI (International System of Units) or the IP (Imperial Units) system.
+ * @param {boolean} return_stress_category - default False.
+ * @param {boolean} limit_inputs - default True.
+ */
+
+export function utci_array(
+  tdb,
+  tr,
+  v,
+  rh,
+  units = "SI",
+  return_stress_category = false,
+  limit_inputs = true,
+) {
+  let kwargs;
+  let ret;
+  if (units.toLowerCase() == "ip") {
+    for (let i = 0; i < tdb.length; i++) {
+      kwargs = {
+        tdb: tdb[i],
+        tr: tr[i],
+        v: v[i],
+      };
+      ret = units_converter(kwargs);
+      tdb[i] = ret["tdb"];
+      tr[i] = ret["tr"];
+      v[i] = ret["v"];
+    }
+  }
+  const eh_pa = tdb.map((_tdb, i) => exponential(_tdb) * (rh[i] / 100.0));
+  const delta_t_tr = tdb.map((_tdb, i) => tr[i] - _tdb);
+
+  const pa = eh_pa.map((_eh_pa) => _eh_pa / 10.0); // convert vapour pressure to kPa
+
+  let utci_approx = tdb.map((_tdb, i) => {
+    return utci_optimized(_tdb, v[i], delta_t_tr[i], pa[i]);
+  });
+
+  // Checks that inputs are within the bounds accepted by the model if not return nan
+  if (limit_inputs) {
+    const tdb_valid = valid_range(tdb, [-50.0, 50.0]);
+    const diff_valid = valid_range(delta_t_tr, [-30.0, 70.0]);
+    const v_valid = valid_range(v, [0.5, 17.0]);
+    for (let i = 0; i < tdb_valid.length; i++) {
+      const all_valid = !(
+        isNaN(tdb_valid[i]) ||
+        isNaN(diff_valid[i]) ||
+        isNaN(v_valid[i])
+      );
+      if (!all_valid) {
+        utci_approx[i] = NaN;
+      }
+    }
+  }
+
+  if (units.toLowerCase() == "ip") {
+    for (let i = 0; i < utci_approx.length; i++) {
+      kwargs = {
+        tmp: utci_approx[i],
+      };
+      utci_approx[i] = units_converter(kwargs, "SI")["tmp"];
+    }
+  }
+
+  utci_approx = utci_approx.map((_utci) => round(_utci, 1));
+  if (return_stress_category) {
+    return {
+      utci: utci_approx,
+      stress_category: mapping_arr(utci_approx, stress_categories),
+    };
+  }
+  return utci_approx;
+}
+
+/**
+ *
+ * @param {number} val
+ * @param {object} categories
+ * @returns {string}
+ */
+function mapping(val, categories) {
+  for (const [key, value] of Object.entries(categories)) {
+    if (val > value[0] && val <= value[1]) {
+      return key;
+    }
+  }
+}
+
+/**
+ *
+ * @param {number[]} val
+ * @param {object} categories
+ * @returns {string[]}
+ */
+function mapping_arr(val, categories) {
+  let ret = [];
+  val.map((_v) => {
+    ret.push(mapping(_v, categories));
+  });
+  return ret;
+}
+
+/**
+ *
+ * @param {number} t_db
+ * @returns {number}
+ */
+function exponential(t_db) {
+  const g = [
+    -2836.5744,
+    -6028.076559,
+    19.54263612,
+    -0.02737830188,
+    0.000016261698,
+    7.0229056 * Math.pow(10.0, -10),
+    -1.8680009 * Math.pow(10.0, -13),
+  ];
+  const tk = t_db + 273.15; // air temp in K
+  let es = 2.7150305 * Math.log1p(tk);
+  g.map((_g, i) => {
```

Comment:
do a for loop. map creates an array so if you don't need to create an array don't use map